pandas/core/indexes/base.py

```diff
@@ -1132,19 +1132,6 @@ def _summary(self, name=None):
             name = type(self).__name__
         return f"{name}: {len(self)} entries{index_summary}"
 
-    def summary(self, name=None):
-        """
-        Return a summarized representation.
-
-        .. deprecated:: 0.23.0
-        """
-        warnings.warn(
-            "'summary' is deprecated and will be removed in a future version.",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self._summary(name)
```

Comment:
Do we need the private method any more? Not super familiar with this one but doesn't seem like something we'd use internally?

Reply:
Yes, we use it in `DataFrame.info`